src/components/inputs/SearchSelect.jsx

```diff
@@ -0,0 +1,427 @@
+    options = _.filter(options, (option) => !(option.isGroup && option.options.length === 0 && hideGroupsWithNoMatch));
+
+    //Filter non grouped options
+    options = _.filter(options, (option) => option.isGroup || (!filterValue || filterOption(filterValue, option)));
+
+    return options;
+  },
+
+  //Element Helpers
+
+  _openList() {
+    this.setState({isFocused: true});
+  },
+
+  _closeList() {
+    this._blurFilterInput();
+    this.setState({isFocused: false});
+    this._closeAllGroups();
+  },
+
+  _focusFilterInput() {
+    this.refs.searchInput.getDOMNode().focus();
+  },
+
+  _blurFilterInput() {
+    this.refs.searchInput.getDOMNode().blur();
+  },
+
+  //Elements Listeners
+
+  _cancelBlurInterval() {
+    if (this._blurInterval) {
+      clearTimeout(this._blurInterval);
+      this._blurInterval = null;
+    }
+  },
+
+  _onFocus(e) {
+    this._cancelBlurInterval();
+    this._openList();
+  },
+
+  _onBlur(e) {
+    this._cancelBlurInterval();
+    this._blurInterval = setTimeout(() => {
+      this._closeList();
+    }, 400);
+  },
+
+  _onSelectValueClick(e) {
+    this._focusFilterInput();
+  },
+
+  _onFilterInputChange(e) {
+    let newValue = e.target.value;
+    let previousValue = this.state.filterValue;
+
+    this._removeSelection();
+    this._updateFilterValue(newValue);
+
+    if (newValue === '') {
+      this._closeAllGroups();
+    }
+
+    let wasEmpty = previousValue.length === 0 && newValue.length > 0;
+    if (wasEmpty) {
+      this._openAllGroups();
+    }
+  },
+
+  _onFilterInputBlur(e) {
+    this._cancelBlurInterval();
+    this._blurInterval = setTimeout(() => {
+      this._clearFilterValue();
+      this._closeList();
+    }, 100);
+  },
+
+  _onFilterInputKeyDown(e) {
+    switch (e.which) {
+    case KEY_CODES.TAB:
+    case KEY_CODES.ENTER:
+      this._selectOption(this.state.suggestedOption);
+      break;
+    case KEY_CODES.ARROW_UP:
+      this._suggestPreviousOption(this.state.filterValue, this.state.suggestedOption);
+      break;
+    case KEY_CODES.ARROW_DOWN:
+      this._suggestNextOption(this.state.filterValue, this.state.suggestedOption);
+      break;
+    }
+  },
+
+  _onGroupClick(group, e) {
+    this._cancelBlurInterval();
+    this._toggleGroupOpenState(group);
+  },
+
+  _onOptionSelect(option, e) {
+    this._selectOption(option);
+  },
+
+  //Renders
+
+  render() {
+    let {
+      className,
+      placeHolder,
+      valueLink: {value},
+    } = this.props;
+    let { isFocused, filterValue } = this.state;
+    let filteredOptions = this._getFilteredOptions(filterValue);
+
+    return (
+      <div
+        className={classNames('Select', `Select--${isFocused ? 'opened' : 'closed'}`, className)}
+        onMouseLeave={this._onBlur}
+      >
+        <div className="Select-input"
+          onClick={this._onSelectValueClick}
+        >
+          <input
+            className={classNames('Select-filterInput', !filterValue && 'Select-filterInput-isEmpty')}
```

Comment:
`isEmpty` is be a modifier, so it should be either `.isEmpty` or `Select-filterInput--empty`.